### PR TITLE
add relationship management

### PR DESCRIPTION
**Pull request by DavidCramer**

### CODE REVIEW

File: php/class-relate.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	 *
+	 * @param int    $attachment_id The attachment ID.
+	 * @param string $version       The version upgrading to.
+	 */
+	public function upgrade_relation( $attachment_id, $version ) {
+		$asset_plugin_version = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['plugin_version'], true );
+		if ( ! empty( $asset_plugin_version ) && version_compare( $asset_plugin_version, $version, '<' ) ) {
+			$meta = get_post_meta( $attachment_id, Sync::META_KEYS['cloudinary'], true );
+			if ( ! empty( $meta[ Sync::META_KEYS['transformation'] ] ) ) {
+				$this->media->update_post_meta( $attachment_id, Sync::META_KEYS['transformation'], $meta[ Sync::META_KEYS['transformation'] ] );
```
Comment (DavidCramer) [inline php/class-relate.php:64]: @pereirinha right here it should unset the transformations and update the post meta.

File: php/class-connect.php [hunk 2]
```diff
@@ -456,7 +456,7 @@ public function check_status() {
 		$status = $this->test_ping();
 		$this->settings->get_setting( 'status' )->save_value( $status );
 
-		return $status;
+		return is_wp_error( $status ) ? $status->get_error_message() : $status['status'];
```
Comment (pereirinha) [inline php/class-connect.php:459]: @DavidCramer 

This change will require a change in the docblock for this method, right?

**Review by pereirinha — CHANGES_REQUESTED** [review]
@DavidCramer 

Back to you.

**Comment by DavidCramer** [comment]
Closing as #773 is cleaner.

**Pull request closed without merging**